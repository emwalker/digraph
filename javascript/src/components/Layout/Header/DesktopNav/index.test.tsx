import React from 'react'
import { shallow } from 'enzyme'

import { UnwrappedDesktopNav as DesktopNav } from './index'

const props = {
  viewer: {
    name: 'Frotz',
    defaultRepository: {
      rootTopic: {
        resourcePath: '/some-repo/topics/123',
      },
    },
  },
}

describe('<DesktopNav />', () => {
  // @ts-expect-error
  const wrapper = shallow(<DesktopNav {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
