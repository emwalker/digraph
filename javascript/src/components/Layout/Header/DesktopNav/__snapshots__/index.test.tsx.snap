// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DesktopNav /> renders 1`] = `
<header
  className="header px-sm-3 px-md-6 px-lg-4"
>
  <nav
    className="navLeft d-inline-block"
  >
    <h1
      className="h3 text-normal"
    >
      <a
        className="primaryLogo n-link no-underline d-flex"
        href="/"
      >
        <div
          className="mr-2 d-inline-block"
        >
          <_default
            fill="#000"
            height="28px"
            width="28px"
          />
        </div>
        Digraph
      </a>
    </h1>
  </nav>
  <div
    className="searchBox"
  >
    <Relay(SearchBox) />
  </div>
  <nav
    className="userNav"
  >
    <a
      className="Link--primary px-2"
      href="https://blog.digraph.app"
    >
      Blog
    </a>
    <Link
      className="Link--primary px-2"
      id="recent-activity"
      to="/recent"
    >
      Recent
    </Link>
    <Link
      className="Link--primary px-2"
      to={
        Object {
          "pathname": "/wiki/topics/df63295e-ee02-11e8-9e36-17d56b662bc8",
          "state": Object {
            "itemTitle": "Everything",
            "orgLogin": "wiki",
            "repoName": "General collection",
          },
        }
      }
    >
      Everything
    </Link>
    <Relay(UserNav)
      viewer={
        Object {
          "defaultRepository": Object {
            "rootTopic": Object {
              "resourcePath": "/some-repo/topics/123",
            },
          },
          "name": "Frotz",
        }
      }
    />
  </nav>
</header>
`;
